notebooks/community/aisamples/AIsample - Uplift Modelling.ipynb

```diff
@@ -0,0 +1,882 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "transformer = (\n",
+    "    Featurize().setOutputCol(\"features\").setInputCols(FEATURE_COLUMNS).fit(raw_df)\n",
+    ")\n",
+    "\n",
+    "df = transformer.transform(raw_df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "train_df, test_df = df.randomSplit([0.8, 0.2], seed=42)\n",
+    "\n",
+    "print(\"Size of train dataset: %d\" % train_df.count())\n",
+    "print(\"Size of test dataset: %d\" % test_df.count())\n",
+    "\n",
+    "train_df.groupby(TREATMENT_COLUMN).count().show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "### Split treatment-control dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "treatment_train_df = train_df.where(f\"{TREATMENT_COLUMN} > 0\")\n",
+    "control_train_df = train_df.where(f\"{TREATMENT_COLUMN} = 0\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "## Step 3: Model Training and Evaluation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "### Uplift Modelling: T-Learner with LightGBM"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def train(train_df):\n",
+    "    classifier = (\n",
+    "        LightGBMClassifier()\n",
+    "        .setFeaturesCol(\"features\")\n",
+    "        .setNumLeaves(10)\n",
+    "        .setNumIterations(100)\n",
+    "        .setObjective(\"binary\")\n",
+    "        .setLabelCol(LABEL_COLUMN)\n",
+    "    )\n",
+    "\n",
+    "    model = classifier.fit(train_df)\n",
+    "\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "treatment_model = train(treatment_train_df)\n",
+    "control_model = train(control_train_df)"
```

Comment:
nit: you dont need the function here, you can make the classifier and it it on each dataframe to yield the two models

Reply:
i think this still stands